File: src/main/java/api/ApiNodeManager.java
package api;

import java.util.ArrayList;
import java.util.Arrays;

class ApiNodeManager {
    private final ArrayList<ApiNode> registeredNodes = new ArrayList<>();

    /**
     * 注册 Api 节点
     *
     * @param nodes Api 节点
     */
    public void register(ApiNode... nodes) {
        registeredNodes.addAll(Arrays.asList(nodes));
    }

    /**
     * 获取 Api 节点
     *
     * @param nodePath Api 节点名
     * @return 匹配到的 Api 节点 或 null
     * @see ApiNode#path()
     */
    public ApiNode getNode(String nodePath) {
        for (ApiNode node : registeredNodes) {
            if (node.path().equalsIgnoreCase(nodePath)) {
                return node;
            }
        }

        return null;
    }
}

File: src/main/kotlin/org/hydev/ApiNodeManager.kt
package org.hydev

/**
 * Api node manager
 *
 * @author HyDEV Team (https://github.com/HyDevelop)
 * @author Hykilpikonna (https://github.com/hykilpikonna)
 * @author Vanilla (https://github.com/VergeDX)
 * @since 2020-07-03 12:34
 */
class ApiNodeManager
{
    private val nodes = ArrayList<ApiNode>()

    /**
     * Register one or many nodes
     *
     * @param nodes API Nodes that you want to register.
     */
    fun register(vararg nodes: ApiNode)
    {
        this.nodes.addAll(nodes)
    }

    /**
     * Fancier way to register a node
     *
     * @param node
     */
    operator fun plusAssign(node: ApiNode)
    {
        register(node)
    }

    /**
     * Get api node
     *
     * @param path Path of the node
     * @return Api node or null
     */
    operator fun get(path: String): ApiNode?
    {
        val lowerPath = path.toLowerCase()

        for (node in nodes)
        {
            if (node.path == lowerPath)
            {
                return node
            }
        }

        return null
    }
}
